ping.go

```diff
@@ -478,6 +487,16 @@ func (p *Pinger) runLoop(
 		case <-timeout.C:
 			return nil
 
+		case <-ptimeout.C:
```

Comment:
This approach has been proposed before. Similar to #176 and #202.

The problem is this only checks for packet timeouts every time the ticker runs, which means the per-packet timeouts are not going to be accurate.

For example, if you have an interval of 1s timeout of 15s, the second packet won't actually timeout until the second interval at 30s. This means it will have timed out at 29 seconds.

I think in order to do this correctly we need to reset the PktTimeout timer each time the oldest packet is received. This way we inch the timeout forward to the next un-received packet.